pkg/pgmodel/cache/series_cache.go

```diff
@@ -26,21 +25,24 @@ const DefaultSeriesCacheSize = 1000000
 // SeriesCache is a cache of model.Series entries.
 type SeriesCache interface {
 	Reset()
-	GetSeriesFromProtos(labelPairs []prompb.Label) (series *model.Series, metricName string, err error)
+	GetSeriesFromProtos(labelPairs []prompb.Label) (series *model.Series, err error)
 	Len() int
 	Cap() int
 	Evictions() uint64
 }
 
 type SeriesCacheImpl struct {
 	*ResizableCache
+	labelsDict  sync.Map
```

Comment:
don't we want to refcount the entries here and delete from the map if all series using a dict entry get evicted. As it is I think the dict size grows in an unbounded way